### PR TITLE
Update DQM definition 

### PR DESCRIPTION
**Pull request by JoelPasvolsky**
Closes https://github.com/dwavesystems/dwave-ocean-sdk/issues/114

### CODE REVIEW

File: docs/concepts/dqm.rst [hunk 1]
```diff
@@ -4,13 +4,31 @@
...
+The discrete quadratic model (DQM) is a polynomial over *discrete* variables, 
+with all terms of degree two or less, where a discrete variable represents some 
+collection of distinct values, such as ``{red, green, blue, yellow}`` or 
+``{3.2, 67}``, which are called the variable's *cases*.  
+
+A discrete quadratic model may be defined as
+
+.. math::
+
+    H(\bf{d}) = \sum_{i} a(\bf{d}_i) + \sum_{i,j} b(\bf{d}_i,\bf{d}_j) + c
```
Comment (mcfarljm) [inline docs/concepts/dqm.rst:16]: I believe that the "a" and "b" functions need subscripts:
```suggestion
    H(\bf{d}) = \sum_{i} a_i(\bf{d}_i) + \sum_{i,j} b_{i,j}(\bf{d}_i,\bf{d}_j) + c
```

File: docs/concepts/dqm.rst [hunk 2]
```diff
@@ -4,13 +4,31 @@
...
+
+where :math:`\bf{d}_i` are the discrete variables, :math:`a()` and :math:`b()`
```
Comment (mcfarljm) [inline docs/concepts/dqm.rst:18]: Perhaps:
```suggestion
where :math:`\bf{d}_i` are the discrete variables, :math:`a_i()` and :math:`b_{i,j}()` 
```
?

**Review by mcfarljm — COMMENTED** [review]
Looks good, a couple minor comments on subscripting the "a" and "b" functions

**Comment by hsadeghidw** [comment]
I have some doubts about the new definition. I think that the new definition can map to a DQM.

We want to make it clear to the users that there are independent linear and quadratic biases for a combination of variables and cases.

I think one thing to add to this [definition](https://docs.ocean.dwavesys.com/en/stable/concepts/dqm.html) is that the variable picks one and only one case. Otherwise, in my opinion, the definition is clear enough.

**Comment by jackraymond** [comment]
On balance I think it would be better to gradually move to a simpler definition; although I appreciate inertia may be a sufficient argument against. My thinking is here:
https://confluence.dwavesys.com/pages/viewpage.action?pageId=89505592

File: docs/concepts/dqm.rst (hunk 1 above)
Comment (jackraymond) [reply docs/concepts/dqm.rst:16]: yes, typo

File: docs/concepts/dqm.rst (hunk 2 above)
Comment (jackraymond) [reply docs/concepts/dqm.rst:18]: agree

**Review by jackraymond — COMMENTED** [review]
Happy with structure implemented by JoelPavolsky, and indexing errors corrected by mcfarljm, looks improved to me.

**Comment by randomir** [comment]
With indexing corrections committed, I gather, @jackraymond, you're fine with merging this?

@hsadeghidw, where do you stand on this? Not sure how to read your comments above.

**Review by randomir — APPROVED** [review]
This looks like an improvement to me.

**Comment by randomir** [comment]
Needless to say, subsequent improvements always welcome in a follow-up PR.

**Pull request merged**